modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php

```diff
@@ -165,7 +169,8 @@ public function getRecipients(array $data, int $last_id, int $limit): array {
           if (
             $group_user !== NULL &&
             $owner_id != $membership->getUser()->id() &&
-            !$this->groupMuteNotify->groupNotifyIsMuted($group, $membership->getUser())
+            !$this->groupMuteNotify->groupNotifyIsMuted($group, $membership->getUser()) &&
+            $entity->access('view', $group_user)
```

Comment:
What is the performance impact of this change for a group of >250 users? How much slower is it?

Reply:
@OH17L We agreed that it would be better to move this access check to separate PR.

Reply:
@OH17L Please see the above comment.

Reply:
I've created a group as SM and added 300+ members directly. Then I logged in as a normal group member and created topics in this group. After creating each topic I run cron manually (the Docker container for cron has been deactivated). Test execution time (in seconds) when the access checker is...

| Present? | Test 1 | Test 2 | Test 3 | Test 4 | Test 5 |
| :---: | ---: | ---: | ---: | ---: | ---: |
| No | 16 | 17 | 17 | 17 | 18 |
| Yes | 19 | 18 | 18 | 24 | 18 |

As you can see the difference is really small so the impact of changes is minimal. So, I'd say that we can use this additional condition for checking access.

Reply:
Hi @bramtenhove, could you check results of @OH17L investigation and tell if you think it's still an issue. 
We already using this patch in project, and don't see any issues with that.

Reply:
Nice work @OH17L! Good to know this isn't an issue 👍 

cc @ribel